### DISCUSSION TITLE
Zuzendu / Bizkaia

### DISCUSSION THREAD

**Discussion by arraintxo** [Barnetik/tbai-php-lib#30] [Ideas]
Bizkaian ez da Zuzendu zerbitzua existitzen. 

Beraien ohiko galderen erantzunagatik ulertzen dudanez, Bizkaian faktura zuzentzaileen eta baliogabetzeen bitartez egin behar da prozesu hau.

https://www.batuz.eus/eu/ohiko-galderak?q=alda+al+daitezke#btz_pregunta_6899939

Nire **kezkak** hurrengoak dira:

1. Zein kasutan erabili behar da Zuzendu eta zein kasutan erabili behar da faktura zuzentzailea/baliogabetzea?
2. Hiru herrialdeetako APIa berbera izan beharko litzateke liburutegian? Nahiz eta Bizkaian ez den Zuzendu zerbitzua existitzen, zerbitzua modu faltsu batean "gaitu" beharko genuke? Nahiz eta atzetik TicketBai faktura zuzentzaile bat bidali?

Bigarren kezkari buruz ez dakit zein punturarte batu genezaken prozesu hau erabiltzailearentzat transparentea izan dadin, Gipuzkoa eta Arabak faktura zuzentzaileak bidaltzea ere baimentzen dutenez, prozesu ezberdinak direla esango nuke, errore ezberdinetatik eratorriak. 

TicketBai Faktura zuzentzaile eta baliogabetzaileak zenbaki berriak sartzen dituzte sisteman eta ulertzen dudanez Zuzenduren kasuan ez da hau gertatzen, beraz baliteke guzti honek zentzurik ez edukitzea. 

Zer deritzozue?

----------------------------------------------

En Bizkaia no existe el servicio Zuzendu.

Por lo que podemos extraer de sus preguntas frecuentes, las correcciones se deben realizar mediante ficheros TicketBai de rectificación y/o anulación.

https://www.batuz.eus/es/preguntas-frecuentes?q=se+pueden+modificar+los+datos+de+una+factura+ya+expedida#btz_pregunta_6899939

Mis **dudas** son las siguientes:

1. ¿En qué casos debe utilizarse el servicio Zuzendu y en cuáles la generación de ficheros TicketBai de rectificación/anulación?
2. ¿La API de la librería debería ser idéntica para los tres territorios? Aunque en Bizkaia no exista Zuzendu, ¿deberíamos "habilitar" un servicio falso que realmente no haga más que emitir facturas rectificativas?

Acerca de la segunda duda, no sé hasta que punto podemos unificar el proceso para que sea transparente para el usuario. Gipuzkoa y Araba permiten emitir facturas rectificativas y anulaciones, por lo que entiendo que la razón para emitir un fichero TicketBai o un fichero Zuzendu deriva de casuísticas diferentes.

Por otro lado los ficheros TicketBai para rectificación y anulación de facturas se emiten con nuevas numeraciones, mientras que en Zuzendu entiendo que esto no ocurre, por lo que puede que toda la idea de unificar las APIs directamente no tenga ningún sentido.

Qué opináis?

**Comment by goear** [comment]
Kaixo @arraintxo, acabo de escribir a ticketbai bizkaia para preguntar los motivos por los cuales no existe el servicio zuzendu en el territorio, y si tienen intención de integrarlo. En cuanto respondan, lo pego aquí.

**Comment by goear** [comment]
@arraintxo han respondido desde Bizkaia:

En Bizkaia no existe el servicio Zuzendu.

Por lo que podemos extraer de sus preguntas frecuentes, las correcciones se deben realizar mediante ficheros TicketBai de rectificación y/o anulación.

https://www.batuz.eus/es/preguntas-frecuentes?q=se+pueden+modificar+los+datos+de+una+factura+ya+expedida#btz_pregunta_6899939

¿En qué casos debe utilizarse el servicio Zuzendu y en cuáles la generación de ficheros TicketBai de rectificación/anulación?
ZUZENDU se puede utilizar para las facturas de operaciones de contribuyentes de Araba y Gipuzkoa, pero no para facturas de operaciones de contribuyentes bizkainos. De hecho, de momento no está contemplado en Bizkaia un servicio equivalente al de ZUZENDU de Gipuzkoa y Araba.

**En el caso de Bizkaia, para poder modificar los datos del fichero TicketBAI, habría dos opciones:**


• Anular la factura mediante el sistema TicketBAI, que pasará a estado “Anulado”, y emitir una nueva factura (que habría que entregar al destinatario y tendría nuevo número y fecha de factura). Esto equivale a lo que hoy en día sería “romper” la factura anterior.

• La otra opción sería emitir una factura rectificativa corrigiendo los errores, aunque los importes fueran los mismos. Esta factura rectificativa también se debería entregar al destinatario y deberá tener una serie distinta.

Recomendamos ver las preguntas frecuentes 19, 20, 21 y los ejemplos de las opciones de rectificación en las preguntas frecuentes 45, 47 a 52, 60, 61, 184 y 188 en la web [Preguntas frecuentes - BATUZ - Bizkaia.eus](https://www.batuz.eus/es/preguntas-frecuentes).

Por otro lado, hay ocasiones en las que la factura que se le entrega al destinatario es correcta y lo que es incorrecto son los datos que se han enviado en el fichero TicketBAI. Por ejemplo, podría tratarse de un error en la clave de IVA (que es un dato que no forma parte del contenido de la factura que se le entrega al cliente pero sí forma parte del fichero TicketBAI). Para ese tipo de operaciones en las que no tiene sentido anular la factura para el destinatario ni rectificar la factura, se pueden realizar anotaciones de “ajuste” en el subcapítulo 1.2 de facturas emitidas sin software garante. Por ejemplo, hay casos en los que un contribuyente debe aplicar la clave 52-Régimen simplificado de IVA en sus facturas, pero por equivocación, ha expedido todas las facturas de un mes o de incluso un año con la clave 01-Operaciones en régimen general. En estos casos, para corregir la situación, se tendría que realizar una anotación en el subcapítulo 1.2 del LROE del contribuyente, en el que se corregirían las claves informadas en el LROE.

**¿La API de la librería debería ser idéntica para los tres territorios? Aunque en Bizkaia no exista Zuzendu, ¿deberíamos "habilitar" un servicio falso que realmente no haga más que emitir facturas rectificativas?**


En Bizkaia no hay un servicio de Zuzendu, por lo que el software que se comercialice en Bizkaia no tendría que incluir dicha opción. El propio software puede decidir qué opciones de corrección de facturas permite. Por ejemplo, hay softwares que realizan las correcciones únicamente por rectificación (sin que exista la opción de anular). Por otro lado, las anotaciones de ajuste de las operaciones informadas incorrectamente en el fichero TicketBAI que se corrigen a través del subcapítulo 1.2-Facturas emitidas sin software garante, tal y como se ha comentado en la respuesta anterior, se pueden hacer manualmente por el usuario mediante la utilización de la aplicación del LROE, es decir, realizando la anotación en el LROE a mano en vez de utilizar los servicios web.

**Comment by arraintxo** [comment]
Podemos entender entonces que no hay necesidad de modificar la librería para esto, ya que el resultado de un caso y otro son diferentes, por lo que por ahora se queda como está.

Eskerrik asko!
